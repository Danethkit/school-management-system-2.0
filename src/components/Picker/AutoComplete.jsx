import React from "react";
import PropTypes from "prop-types";
import deburr from "lodash/deburr";
import Downshift from "downshift";
import { makeStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Paper from "@material-ui/core/Paper";
import MenuItem from "@material-ui/core/MenuItem";
import InputAdornment from '@material-ui/core/InputAdornment'
import Popper from '@material-ui/core/Popper';
import Fade from '@material-ui/core/Fade';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';


function renderInput(inputProps) {
    const { InputProps, classes, ref,label,onClick,width, ...other } = inputProps;
    return (
        <TextField
            // disabled={readonly}
            style={{minWidth:width}}
            variant='outlined'
            InputProps={{
                inputRef: ref,
                startAdornment: <InputAdornment position="start"><b>{label}</b></InputAdornment>,
                classes: {
                    root: classes.inputRoot,
                    input: classes.inputInput
                },
                ...InputProps
            }}
            {...other}
            onClick={onClick}
        />
    );
}

function renderSuggestion(suggestionProps) {
    const {
        suggestion,
        index,
        itemProps,
        highlightedIndex,
        selectedItem,
        rest,
        selectedFaculty
    } = suggestionProps
    let selectedFacultyArray = []
    let isBusy = false
    if('col' in rest && Object.keys(selectedFaculty).length !== 0){
      let {row, col, weekStr} = rest
      for(let course in selectedFaculty['res'][weekStr]){
        for(let batch in selectedFaculty['res'][weekStr][course]){
          for(let semester in selectedFaculty['res'][weekStr][course][batch]){
            for(let group in selectedFaculty['res'][weekStr][course][batch][semester]){
              if(col in selectedFaculty['res'][weekStr][course][batch][semester][group]){
                if(row in selectedFaculty['res'][weekStr][course][batch][semester][group][col]){
                  selectedFacultyArray.push(selectedFaculty['res'][weekStr][course][batch][semester][group][col][row])
                }
              }
            }
          }
        }
      }
      if(selectedFacultyArray.length !== 0){
          selectedFacultyArray.forEach(e=> {
              if(!e) return
              if(!isBusy){
                  if(selectedItem){
                      selectedFacultyArray.forEach(e => isBusy = selectedItem.includes(e.split('(')[1]))
                  } else{
                      isBusy = suggestion.label.includes(e.split('(')[1])
                  }
              }
          })
      }
    }

    const isHighlighted = highlightedIndex === index;
    const isSelected = (selectedItem || "").indexOf(suggestion.label) > -1;
    return (
        <MenuItem
            {...itemProps}
            key={suggestion.label}
            selected={isHighlighted}
            disabled={isBusy}
            component="div"
            style={{
                fontWeight: isSelected ? 500 : 400
            }}
        >
            {suggestion.label}
        </MenuItem>
    );
}

function getSuggestions(value, { showEmpty = false } = {}, suggestions) {
    const inputValue = deburr(value.trim()).toLowerCase();
    const inputLength = inputValue.length;
    return inputLength === 0 && !showEmpty
        ? []
        : suggestions.filter(suggestion => suggestion.label.toLowerCase().includes(inputValue));
}

const useStyles = makeStyles(theme => ({
        root: {
            flexGrow: 1,
        },
        container: {
            flexGrow: 1,
            position: "relative"
        },
        paper: {
            position: "absolute",
            zIndex: 999,
            marginTop: theme.spacing(1),
            left: 0,
            right: 0,
            width:400,
            boxShadow: theme.shadows[5]
        },
        inputRoot: {
            flexWrap: "wrap"
        },
        inputInput: {
            width: "auto",
            flexGrow: 1,
        },
        divider: {
            height: theme.spacing(2)
        }
    })
);

function AutoComplete({suggestions, value=null, onChange, label, selectedFaculty,width, ...rest}) {
  console.log('render');
    const classes = useStyles();
    const [anchorEl, setAnchorEl] = React.useState(null);
    function handleClose(event) {
        setAnchorEl(null);
    }
    function handleClick(event) {
        setAnchorEl(anchorEl ? null : event.currentTarget);
    }

    const open = Boolean(anchorEl);
    const id = open ? 'simple-popper' : null;

    // if(rest.disabled){
    //   value = 'Holiday'
    // }
    return  <Downshift onChange={onChange} selectedItem={value}>
        {downshift => {
            const {onBlur, onChange, onFocus, ...inputProps} = downshift.getInputProps({
                onChange: event => {
                    if (event.target.value === "") {
                        downshift.clearSelection()
                    }
                },
                onFocus: downshift.openMenu,
            })
            return <div className={classes.container}>
                {
                    renderInput({
                        fullWidth:true,
                        width:width,
                        classes,
                        InputLabelProps: downshift.getLabelProps({shrink:true}),
                        InputProps: {onBlur, onChange, onFocus,onClick:handleClick},
                        inputProps,
                        label,
                        // readonly: rest.disabled
                    })
                }
                <ClickAwayListener onClickAway={handleClose} >
                    <Popper id={id} open={open} anchorEl={anchorEl} transition >
                        {({ TransitionProps }) => (
                            <Fade {...TransitionProps}>
                                <Paper style={{minWidth:225}}>
                                    {getSuggestions(downshift.inputValue, { showEmpty: true }, suggestions.map(e=>({label:e}))).map(
                                        (suggestion, index) =>
                                            renderSuggestion({
                                                suggestion,
                                                index,
                                                itemProps: downshift.getItemProps({ item: suggestion.label }),
                                                highlightedIndex: downshift.highlightedIndex,
                                                selectedItem: value,
                                                selectedFaculty,
                                                rest
                                            })
                                    )}
                                </Paper>
                            </Fade>
                        )}
                    </Popper>
                </ClickAwayListener>
            </div>
        }}
    </Downshift>
}

export default AutoComplete
renderSuggestion.propTypes = {
    highlightedIndex: PropTypes.number,
    index: PropTypes.number,
    itemProps: PropTypes.object,
    selectedItem: PropTypes.string,
    suggestion: PropTypes.shape({ label: PropTypes.string }).isRequired
};