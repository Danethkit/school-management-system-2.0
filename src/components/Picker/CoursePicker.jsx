import React, { Component } from "react";
import {
  MenuItem,
  TextField,
  InputAdornment,
  withStyles
} from "@material-ui/core/";
import classNames from "classnames";

const styles = theme => ({
  margin: {
    margin: theme.spacing.unit
  },
  textFile: {
    fontSize: 14
  }
});

let id = 0;

function inputCourse(id, course) {
  id += 1;
  return { id, course };
}
const courseData = [
  inputCourse(id, "Software Engineering"),
  inputCourse(id, "Hospitality Management")
];

class CoursePicker extends Component {
  handleChange = prop => event => {
    this.setState({ [prop]: event.target.value });
  };
  state = {
    Course: ""
  };

  render() {
    const { classes } = this.props;
    return (
      <div>
        <TextField
          select
          fullWidth
          autoFocus
          className={classNames(classes.margin)}
          value={this.state.Course}
          onChange={this.handleChange("Course")}
          InputProps={{
            startAdornment: (
              <InputAdornment
                disableTypography={true}
                className={classes.textFile}
                position="start"
              >
                <b>Course:</b>{" "}
              </InputAdornment>
            )
          }}
        >
          {courseData.map(option => (
            <MenuItem key={option.id} value={option.course}>
              {option.course}
            </MenuItem>
          ))}
        </TextField>
      </div>
    );
  }
}

export default withStyles(styles)(CoursePicker);
