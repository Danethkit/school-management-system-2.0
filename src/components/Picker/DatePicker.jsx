import "date-fns";
import React from "react";
import PropTypes from "prop-types";
import Grid from "@material-ui/core/Grid";
import { withStyles } from "@material-ui/core/styles";
import DateFnsUtils from "@date-io/date-fns";
import { MuiPickersUtilsProvider, DatePicker } from "material-ui-pickers";

const styles = {
  grid: {
    width: "20%",
    paddingTop: 10
  }
};

class datePicker extends React.Component {
  state = {
    selectedDate: new Date()
  };

  handleDateChange = date => {
    this.setState({ selectedDate: date });
  };

  render() {
    const { classes } = this.props;

    return (
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <Grid container>
          <Grid item sm>
            Date :
          </Grid>
          <Grid item sm={9}>
            
              <DatePicker
                value={this.state.selectedDate}
                onChange={this.handleDateChange}
                className={classes.filledInput}
              />
            
          </Grid>
        </Grid>
      </MuiPickersUtilsProvider>
    );
  }
}

datePicker.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(datePicker);
