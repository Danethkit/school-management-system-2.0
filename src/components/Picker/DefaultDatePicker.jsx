import "date-fns";
import React from "react";
import DateFnsUtils from "@date-io/date-fns";
import { InputAdornment } from '@material-ui/core'
import { MuiPickersUtilsProvider, KeyboardDatePicker } from "@material-ui/pickers";

export default ({value, onChange , label}) =>{
    return (
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <KeyboardDatePicker
          format="MM/dd/yyyy"
          autoOk
          maxDate ={new Date()}
          inputVariant='outlined'
          autoFocus
          fullWidth
          variant="inline"
          value={value}
          onChange={onChange}
          style={{marginTop:16, marginBottom:4}}
          InputProps={{
            startAdornment: (
              <InputAdornment
                disableTypography={true}
                position="start"
                style={{fontSize:14,}}
              >
                <b>{label}:</b>
              </InputAdornment>
            )
          }}
        />
      </MuiPickersUtilsProvider>
    );
}


