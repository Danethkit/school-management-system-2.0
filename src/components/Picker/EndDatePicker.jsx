import "date-fns";
import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import DateFnsUtils from "@date-io/date-fns";
import { MuiPickersUtilsProvider, DatePicker } from "material-ui-pickers";
import { InputAdornment } from "@material-ui/core";
import classNames from "classnames";

const styles = theme => ({
    margin: {
        margin: theme.spacing.unit
    },
    textFile: {
        fontSize: 14
    }
});

class EndDatePicker extends React.Component {
    state = {
        selectedDate: new Date()
    };

    handleDateChange = date => {
        this.setState({ selectedDate: date });
    };

    render() {
        const { classes } = this.props;

        return (
            <MuiPickersUtilsProvider utils={DateFnsUtils}>
                <DatePicker
                    autoFocus
                    fullWidth
                    value={this.state.selectedDate}
                    onChange={this.handleDateChange}
                    className={classNames(classes.margin, classes.textField)}
                    InputProps={{
                        startAdornment: (
                            <InputAdornment
                                disableTypography={true}
                                className={classes.textFile}
                                position="start"
                            >
                                <b>EndDate:</b>
                            </InputAdornment>
                        )
                    }}
                />
            </MuiPickersUtilsProvider>
        );
    }
}

EndDatePicker.propTypes = {
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(EndDatePicker);