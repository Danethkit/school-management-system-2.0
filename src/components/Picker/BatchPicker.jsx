import React, {useEffect} from "react";
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { onBatchChange, onSemesterChange } from '../../redux/ActionCreator/userBehavior'
import AutoComplete from './AutoComplete'

const BatchPicker = ({dispatch, uid, batch, subjectInfo, userIden, course}) => {
  let data = {}
  if(uid.uid === 'admin'){
    data = subjectInfo
  }else {
    data = userIden
  }
  useEffect(()=>{
    if(!batch){
      dispatch(onSemesterChange(null))
    }
  })
  let allBatch = []
  try{
    let batches = data[course]
    allBatch = Object.keys(batches).sort((a, b)=> a.split(" ")[1] - b.split(" ")[1])
  }catch(err){}
  let action = bindActionCreators(onBatchChange, dispatch)

  return <AutoComplete
          value={batch}
          onChange={action}
          label = "Batch"
          suggestions={allBatch} />
}

export default connect(state => ({
  batch:state.changePicker.batch,
  course: state.changePicker.course,
  subjectInfo: state.initData.subjectInfo,
  uid: state.initData.uid,
}))(BatchPicker)