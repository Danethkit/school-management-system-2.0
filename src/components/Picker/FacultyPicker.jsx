import React, {useEffect} from "react";
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { onFacultyChange } from '../../redux/ActionCreator/userBehavior'
import AutoComplete from './AutoComplete'


const FacultyPicker = ({dispatch, faculty, course, batch, semester, group, session, userIden}) => {
  const uid = localStorage.getItem('uid')
  let actions = bindActionCreators(onFacultyChange, dispatch)
  let suggestions = []
  let temp = {}
  if(uid == 1){
    try{
      temp = userIden[course][batch][semester][group]
      for (let e in temp) {
        if (temp[e]['faculty'] && !(suggestions.includes(temp[e]['faculty']))) {
          suggestions.push(temp[e]['faculty']);
        }
      }
    }catch(e){}
  }else {
    suggestions.push(userIden['user'])
    faculty = userIden['user']
  }

  useEffect(()=>{
    for(let i in temp){
      if(i === session){
        actions(temp[i]['faculty'])
        break
      }
    }
  }, [session, userIden])
  return <AutoComplete
          value ={faculty}
          onChange ={actions.onFacultyChange}
          label = "Faculty"
          suggestions = {suggestions}
          disable
        />
}
export default connect(state => ({
  faculty:state.changePicker.faculty,
  session:state.changePicker.session,
  course:state.changePicker.course,
  batch:state.changePicker.batch,
  group:state.changePicker.group,
  semester:state.changePicker.semester,
}))(FacultyPicker)