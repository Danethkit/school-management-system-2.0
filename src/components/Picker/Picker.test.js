import React from 'react'
import {shallow} from 'enzyme'
import AutoComplete from './AutoComplete'
import DefaultPicker from './DefaultPicker'
import BatchPicker from './BatchPicker'
import CoursePicker from './CoursePicker'
import DatePicker from './DatePicker'
import DefaultDatePicker from './DefaultDatePicker'
import EndDatePicker from './EndDatePicker'
import FacultyPicker from './FacultyPicker'
import GroupPicker from './GroupPicker'
import SemesterPicker from './SemesterPicker'
import SessionPicker from './SessionPicker'
import StartDatePicker from './StartDatePicker'
import SubjectPicker from './SubjectPicker'
import WeekPicker from './WeekPicker'
describe('Pickers', () => {
    describe('AutoComplete', () => {
        let Auto_Complete
        beforeEach(()=>{
            Auto_Complete = shallow(<AutoComplete/>)
        })
        it('should render without errors', () => {
            expect(Auto_Complete.find('Downshift').length).toBe(1)
        });
    });
    describe('DefaultPicker', () => {
        let Default_Picker
        beforeEach(()=>{
            Default_Picker = shallow(<DefaultPicker/>)
        })
        it('should render without errors', () => {
            //expect(Default_Picker.dive().find('TextField').length).toBe(1)
            expect(Default_Picker).toMatchSnapshot()
        });
    });
    describe('BatchPicker', () => {
        let Batch_Picker
        beforeEach(()=>{
            Batch_Picker = shallow(<BatchPicker/>)
        })
        it('should render without erros', () => {
            expect(Batch_Picker).toMatchSnapshot()
           
        });
    });
    describe('course picker', () => {
       let Course_Picker
        beforeEach(()=>{
            Course_Picker = shallow(<CoursePicker/>)
        })
        it('should render without errors', () => {
            expect(Course_Picker).toMatchSnapshot()
        });
    });
    describe('DatePicker', () => {
        let Date_Picker
        beforeEach(()=>{
            Date_Picker = shallow(<DatePicker/>)
        })
        it('should render without errors', () => {
            expect(Date_Picker).toMatchSnapshot()
        });
    });
    describe('DefaultDatePicker', () => {
        let Default_DatePicker
        beforeEach(()=>{
            Default_DatePicker = shallow(<DefaultDatePicker/>)
        })
        it('should render without errors', () => {
            expect(Default_DatePicker.find('MuiPickersUtilsProvider').length).toBe(1)
            
        });
    });
    describe('EndDatePicker', () => {
        let EndDate_Picker
        beforeEach(()=>{
            EndDate_Picker = shallow(<EndDatePicker/>)
        })
        it('should render without errors', () => {
            expect(EndDate_Picker).toMatchSnapshot()
        });
    });
    describe('FacultyPicker', () => {
        let Faculty_Picker
        beforeEach(()=>{
            Faculty_Picker = shallow(<FacultyPicker/>)
        })
        it('should render without errors', () => {
            expect(Faculty_Picker).toMatchSnapshot()
        });
    });
    describe('GroupPicker', () => {
        let Group_Picker
        beforeEach(()=>{
         Group_Picker = shallow(<GroupPicker/>)
        })
        it('should render without errors', () => {
            expect(Group_Picker).toMatchSnapshot()
        });
    });
    describe('SemesterPicker', () => {
        let Semester_Picker
        beforeEach(()=>{
            Semester_Picker = shallow(<SemesterPicker/>)
        })
        it('should render without errors', () => {
            expect(Semester_Picker).toMatchSnapshot()
        });
    });
    describe('StartDatePicker', () => {
        let StartDate_Picker
        beforeEach(()=>{
            StartDate_Picker = shallow(<StartDatePicker/>)
        })
        it('should render without errors', () => {
            expect(StartDate_Picker).toMatchSnapshot()
        });
    });
    describe('SubjectPicker', () => {
        let Subject_Picker
        beforeEach(()=>{
            Subject_Picker = shallow(<SubjectPicker/>)
        })
        it('should render without errors', () => {
            expect(Subject_Picker).toMatchSnapshot()
        });
    });
    describe('WeekPicker', () => {
        let Week_Picker
        beforeEach(()=>{
            Week_Picker = shallow(<WeekPicker/>)
        })
        it('should render without errors', () => {
            expect(Week_Picker).toMatchSnapshot()
        });
    });
});