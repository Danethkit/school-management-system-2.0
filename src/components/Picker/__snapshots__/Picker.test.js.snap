// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pickers BatchPicker should render without erros 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(BatchPicker) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Pickers DatePicker should render without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(AttendanceDate) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Pickers DefaultDatePicker should render without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WrappedKeyboardPicker
        InputProps={
          Object {
            "startAdornment": <ForwardRef(WithStyles)
              disableTypography={true}
              position="start"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              <b>
                :
              </b>
            </ForwardRef(WithStyles)>,
          }
        }
        allowKeyboardControl={true}
        autoFocus={true}
        autoOk={true}
        format="MM/dd/yyyy"
        fullWidth={true}
        inputVariant="outlined"
        invalidDateMessage="Invalid Date Format"
        maxDate={2019-07-31T06:46:06.472Z}
        maxDateMessage="Date should not be after maximal date"
        minDate={1900-01-01T00:00:00.000Z}
        minDateMessage="Date should not be before minimal date"
        openTo="date"
        variant="inline"
        views={
          Array [
            "year",
            "date",
          ]
        }
      />,
      "style": Object {
        "width": 280,
      },
      "utils": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "InputProps": Object {
          "startAdornment": <ForwardRef(WithStyles)
            disableTypography={true}
            position="start"
            style={
              Object {
                "fontSize": 14,
              }
            }
          >
            <b>
              :
            </b>
          </ForwardRef(WithStyles)>,
        },
        "allowKeyboardControl": true,
        "autoFocus": true,
        "autoOk": true,
        "format": "MM/dd/yyyy",
        "fullWidth": true,
        "inputVariant": "outlined",
        "invalidDateMessage": "Invalid Date Format",
        "maxDate": 2019-07-31T06:46:06.472Z,
        "maxDateMessage": "Date should not be after maximal date",
        "minDate": 1900-01-01T00:00:00.000Z,
        "minDateMessage": "Date should not be before minimal date",
        "onChange": undefined,
        "openTo": "date",
        "value": undefined,
        "variant": "inline",
        "views": Array [
          "year",
          "date",
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WrappedKeyboardPicker
          InputProps={
            Object {
              "startAdornment": <ForwardRef(WithStyles)
                disableTypography={true}
                position="start"
                style={
                  Object {
                    "fontSize": 14,
                  }
                }
              >
                <b>
                  :
                </b>
              </ForwardRef(WithStyles)>,
            }
          }
          allowKeyboardControl={true}
          autoFocus={true}
          autoOk={true}
          format="MM/dd/yyyy"
          fullWidth={true}
          inputVariant="outlined"
          invalidDateMessage="Invalid Date Format"
          maxDate={2019-07-31T06:46:06.472Z}
          maxDateMessage="Date should not be after maximal date"
          minDate={1900-01-01T00:00:00.000Z}
          minDateMessage="Date should not be before minimal date"
          openTo="date"
          variant="inline"
          views={
            Array [
              "year",
              "date",
            ]
          }
        />,
        "style": Object {
          "width": 280,
        },
        "utils": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "InputProps": Object {
            "startAdornment": <ForwardRef(WithStyles)
              disableTypography={true}
              position="start"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              <b>
                :
              </b>
            </ForwardRef(WithStyles)>,
          },
          "allowKeyboardControl": true,
          "autoFocus": true,
          "autoOk": true,
          "format": "MM/dd/yyyy",
          "fullWidth": true,
          "inputVariant": "outlined",
          "invalidDateMessage": "Invalid Date Format",
          "maxDate": 2019-07-31T06:46:06.472Z,
          "maxDateMessage": "Date should not be after maximal date",
          "minDate": 1900-01-01T00:00:00.000Z,
          "minDateMessage": "Date should not be before minimal date",
          "onChange": undefined,
          "openTo": "date",
          "value": undefined,
          "variant": "inline",
          "views": Array [
            "year",
            "date",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Pickers DefaultPicker should render without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef(WithStyles) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "margin": "DefaultPicker-margin-7",
        "textFile": "DefaultPicker-textFile-8",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "margin": "DefaultPicker-margin-7",
          "textFile": "DefaultPicker-textFile-8",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Pickers course picker should render without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(CoursePicker) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
